cli/azd/pkg/commands/pipeline/azdo_provider_test.go

```diff
@@ -0,0 +1,168 @@
+		require.EqualValues(t, testRepoName, details.repoName)
+		require.EqualValues(t, false, details.pushStatus)
+	})
+
+	t.Run("non azure devops https remote", func(t *testing.T) {
+		//arrange
+		provider := &AzdoScmProvider{}
+		ctx := context.Background()
+
+		//act
+		details, e := provider.gitRepoDetails(ctx, "https://github.com/Azure/azure-dev.git")
+
+		//asserts
+		require.Error(t, e, ErrRemoteHostIsNotAzDo)
+		require.EqualValues(t, (*gitRepositoryDetails)(nil), details)
+	})
+
+	t.Run("non azure devops git remote", func(t *testing.T) {
+		//arrange
+		provider := &AzdoScmProvider{}
+		ctx := context.Background()
+
+		//act
+		details, e := provider.gitRepoDetails(ctx, "git@github.com:Azure/azure-dev.git")
+
+		//asserts
+		require.Error(t, e, ErrRemoteHostIsNotAzDo)
+		require.EqualValues(t, (*gitRepositoryDetails)(nil), details)
+	})
+}
+
+func Test_azdo_provider_preConfigureCheck(t *testing.T) {
+	t.Run("accepts a PAT via system environment variables", func(t *testing.T) {
+		// arrange
+		testPat := "12345"
+		provider := getEmptyAzdoScmProviderTestHarness()
+		os.Setenv(azdo.AzDoEnvironmentOrgName, "testOrg")
+		os.Setenv(azdo.AzDoPatName, testPat)
+		testConsole := console.NewMockConsole()
+		ctx := context.Background()
+
+		// act
+		e := provider.preConfigureCheck(ctx, testConsole)
+
+		// assert
+		require.NoError(t, e)
+
+		//cleanup
+		os.Unsetenv(azdo.AzDoPatName)
+	})
+
+	t.Run("returns an error if no pat is provided", func(t *testing.T) {
+		// arrange
+		os.Unsetenv(azdo.AzDoPatName)
+		os.Setenv(azdo.AzDoEnvironmentOrgName, "testOrg")
+		provider := getEmptyAzdoScmProviderTestHarness()
+		testConsole := console.NewMockConsole()
+		testConsole.WhenConfirm(func(options input.ConsoleOptions) bool {
+			return options.Message == "Save the PAT to the  environment file (.env)?"
+		}).Respond(true)
+		testPat := "testPAT12345"
+		testConsole.WhenPrompt(func(options input.ConsoleOptions) bool {
+			return options.Message == "Personal Access Token (PAT):"
+		}).Respond(testPat)
+		ctx := context.Background()
+
+		// act
+		e := provider.preConfigureCheck(ctx, testConsole)
+
+		// assert
+		require.Nil(t, e)
+		require.EqualValues(t, testPat, provider.Env.Values[azdo.AzDoPatName])
+	})
+
+}
+
+func Test_saveEnvironmentConfig(t *testing.T) {
+	tempDir := t.TempDir()
+
+	t.Run("saves to environment file", func(t *testing.T) {
+		// arrange
+		key := "test"
+		value := "12345"
+		provider := getEmptyAzdoScmProviderTestHarness()
+		envPath := path.Join(tempDir, ".test.env")
+		provider.Env = environment.EmptyWithFile(envPath)
+		// act
+		e := provider.saveEnvironmentConfig(key, value)
+		// assert
+		writtenEnv, err := environment.FromFile(envPath)
+		require.NoError(t, err)
+
+		require.EqualValues(t, writtenEnv.Values[key], value)
+		require.NoError(t, e)
+	})
+
+}
+func getEmptyAzdoScmProviderTestHarness() *AzdoScmProvider {
+	return &AzdoScmProvider{
+		Env: &environment.Environment{
```

Comment:
You can use `environment.Ephemeral` for this now, and as we evolve the shape of Env we won't have to go back and fix this up if there's other state that must be initialized.

Reply:
https://github.com/Azure/azure-dev/issues/756